charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml

```diff
@@ -356,7 +356,16 @@ spec:
 {{- if not .Values.prometheus.agentMode }}
 {{- if .Values.prometheus.prometheusSpec.thanos }}
   thanos:
-{{ toYaml .Values.prometheus.prometheusSpec.thanos | indent 4 }}
+{{- omit .Values.prometheus.prometheusSpec.thanos "objectStorageConfig" | toYaml | indent 4 }}
```

Comment:
Why are you using omit ?

Reply:
- I remove `objectStorageConfig` key from being pulled in directly from the `.Values.prometheus.prometheusSpec.thanos` block (as our structure defined in `values.yaml` is not syntactically correct for the thanosconfig CRD
- Dump the rest of the block config as per previous behaviour
- `.Values.prometheus.prometheusSpec.thanos.objectStorageConfig` is set, then convert it from the values syntax to thanos CRD correct syntax